### PR TITLE
Notifications row inline action groundwork

### PR DESCRIPTION
**Pull request by salimbraksa**
Part of #22473

This PR introduces the following updates:
- Add `InlineAction` type into `NotificationsTableViewCellContent.Style`.
- Dynamic notification row alignment:
  - Views align at the top when the text height surpasses the avatar's height.
  - In other cases, views are aligned at the center.

| Before | After |
| ------ | ------ |
|  |  |

## Test Instructions
1. Navigate to Notifications
2. **Verify** the row views are top aligned when the text height is higher than avatar's height
3. Rotate the device.
4. **Verify** the row views are center aligned when the text height is lower than avatar's height

## Regression Notes
1. Potential unintended areas of impact
N/A

5. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

6. What automated tests I added (or what prevented me from doing so)
N/A

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift [hunk 1]
```diff
@@ -21,6 +22,11 @@ struct NotificationsTableViewCellContent: View {
         case altered(Altered)
     }
 
+    struct InlineAction {
+        let icon: SwiftUI.Image
+        let action: () -> Void
+    }
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift:29]: The reason the `icon` type is an `Image`, is to enable creating icons from different bundles:
- Icon from DesignSystem module: `Image.DS.icon(named:...)`
- System Icon: `Image(systemName: ... )`
- WordPress Module: `Image("...")`

**Review by alpavanoglu — APPROVED** [review]
LG2M 🚢

File: WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift [hunk 2]
```diff
@@ -180,26 +180,36 @@ fileprivate extension NotificationsTableViewCellContent {
 
 // MARK: - Helpers
 
-private struct SizeCalculator: ViewModifier {
+private struct SizePreferenceKey: PreferenceKey {
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift:183]: Credits to the author of [this post](https://swiftwithmajid.com/2020/01/15/the-magic-of-view-preferences-in-swiftui/) for this solution

**Comment by alpavanoglu** [comment]
Tested it again on landscape and portrait, all looks well now.

File: WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift [hunk 3]
```diff
@@ -42,22 +49,39 @@ struct NotificationsTableViewCellContent: View {
...
+
         private let info: Style.Regular
 
         fileprivate init(info: Style.Regular) {
             self.info = info
         }
 
         var body: some View {
-            HStack(alignment: .top, spacing: 0) {
+            HStack(alignment: rootStackAlignment, spacing: 0) {
```
Comment (salimbraksa) [inline WordPress/Classes/ViewRelated/Views/List/NotificationsList/NotificationsTableViewCellContent.swift:67]: This is the fix @hassaanelgarem, a dynamic alignment value depending on avatars and texts height

**Pull request merged**